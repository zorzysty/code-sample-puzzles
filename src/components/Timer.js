import * as PropTypes from 'prop-types';
import React from 'react';


const Timer = ({ time, error }) => (
  <div className={`Timer ${error ? 'Timer--error' : ''}`}>
    {time > 0 ? `TIME: ${time}` : 'Drag first item to start the clock.'}
  </div>
);

Timer.propTypes = {
  time: PropTypes.number.isRequired,
  error: PropTypes.bool.isRequired,
};

export default Timer;
