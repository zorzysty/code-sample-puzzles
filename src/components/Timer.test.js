import React from 'react';
import Timer from 'components/Timer';
import renderer from 'react-test-renderer';


describe('Timer component', () => {
  it('should render correctly based on passed props', () => {
    const propsArray = [
      {
        time: 12345,
        error: false,
      },
      {
        time: 15,
        error: true,
      },
      {
        time: 0,
        error: false,
      },
    ];

    propsArray.forEach((props) => {
      const component = renderer.create(<Timer {...props} />);
      const tree = component.toJSON();
      expect(tree)
        .toMatchSnapshot();
    });
  });
});
