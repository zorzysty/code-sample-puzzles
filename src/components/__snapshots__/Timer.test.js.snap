// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer component should render correctly based on passed props 1`] = `
<div
  className="Timer "
>
  TIME: 12345
</div>
`;

exports[`Timer component should render correctly based on passed props 2`] = `
<div
  className="Timer Timer--error"
>
  TIME: 15
</div>
`;

exports[`Timer component should render correctly based on passed props 3`] = `
<div
  className="Timer "
>
  Drag first item to start the clock.
</div>
`;
